crates/runtime/src/lib.rs

```diff
@@ -121,6 +154,31 @@ impl WasmtimeBuildable for Builder {
     }
 }
 
+fn build_wasi_context(io_redirects: Option<IORedirects>) -> Result<WasiCtx> {
```

Comment:
Although this func is named as "build_wasi_context", it is heavily funcitoned as a unit of building io pipes as its argument suggested so. Not a big deal but I would suggest to change it to something that takes a `WasiCtxBuilder` as an argument, and add `stdout/in/err` to the builder, and name the function to `add_io_redirecfts_to_wasi_context`.

Reply:
`fn add_io_redirecfts_to_wasi_context(mut WasiCtxBuilder) -> Result<WasiCtxBuilder>`
`fn build_wasi_context(mut WasiCtxBuilder) -> Result<WasiCtx>`

split the logic to these two functions and use the [Builder pattern](https://rust-unofficial.github.io/patterns/patterns/creational/builder.html)